### PR TITLE
Rust 1.57

### LINKED ISSUE #585: Update to Rust 1.57.0!

### PR DESCRIPTION
**Pull request by ojeda**
Please see the individual commits.

I am particularly curious why `vec!` was compiling in 1.56.0 (@bjorn3?).

v2:
  - `build_error` updated here too: https://github.com/Rust-for-Linux/linux/pull/586/commits/cb94e38bef72bad574293e08c31f583447108ad9
  - `vec!` disabled: https://github.com/Rust-for-Linux/linux/pull/586/commits/2d5e1de8269ee106e0d8343aed0fd2ed695a540d

@wedsonaf Please take a look at the `sysctl` change.

Fixes https://github.com/Rust-for-Linux/linux/issues/585.

### CODE REVIEW

**Comment by bjorn3** [comment]
> In particular, I am particularly curious why vec! was compiling in 1.56.0 (@bjorn3?).

Probably caused by https://github.com/rust-lang/rust/pull/89030 and https://github.com/rust-lang/rust/pull/89036. I suspect previously it would have failed to compile if the function was actually codegened, just not when defined. Is `Sysctl::register` used anywhere?

**Comment by ojeda** [comment]
That seems to be the case indeed -- I just force-called it from 1.56.0 and got the error. Thanks!

File: rust/build_error.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@
 //! functions could still be called in the runtime).
 
 #![no_std]
-#![feature(const_panic, core_panic)]
+#![feature(core_panic)]
```
Comment (nbdd0121) [inline rust/build_error.rs]: rust-lang/rust#88954 is in Rust 1.57 so `core_panic` isn't needed anymore. The panic below can be written as `panic!("{}", msg);`

Comment (ojeda) [reply rust/build_error.rs]: Yeah, @bjorn3 also mentioned it in the issue. Let's do also the `vec!` disabling here then.

**Review by bjorn3 — APPROVED** [review]

**Review by nbdd0121 — APPROVED** [review]

**Review by wedsonaf — APPROVED** [review]
sysctl changes lgtm.

**Review by alex — APPROVED** [review]

**Comment by ojeda** [comment]
Thanks everyone!

**Pull request merged**